### PR TITLE
Fixing Form location when launched on secondary monitor

### LINKED ISSUE #5729: Positioning of the Form on non primary monitor is not centered.
* .NET Core Version: 
.NET 7.0

* Have you experienced this same bug with .NET Framework?: 
NO but .NET framework does not support this scenario.

**Problem description:**
Form directly launched on secondary monitor with different DPI settings than the primary monitor, does not position the form centered.

**Minimal repro:**

-  A simple app on a dual monitor setup with different DPI settings should help repro this.
[WinformsDpiTests.zip](https://github.com/dotnet/winforms/files/7138444/WinformsDpiTests.zip)

### PR DESCRIPTION
**Pull request by dreddy-work**
Fixes #5729.

Control gets location information form `CreateParams `but these were calculated
before `handle `created for the control. In case of launching the `Form` on secondary monitor, DPI
is evaluated only after `handle `is created for the `Form` and the `Form` resized according to the new DPI. Hence, `Form `location need to be recalculated with new bounds information.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5756)

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Any thoughts on how we can test this?

**Comment by dreddy-work** [comment]
> Any thoughts on how we can test this?
From unit test perspective, I didn't figure out on how to simulate this code path. Only manual validation for now.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 1]
```diff
@@ -7813,6 +7813,14 @@ protected virtual void OnHandleCreated(EventArgs e)
                         }
 
                         RescaleConstantsForDpi(old, _deviceDpi);
+
+                        // If control is top-level window and control position is not a default location,
+                        // resizing the control would need location recalculated.
+                        // ex: FormStartPosition.CenterParent or FormStartPosition.CenterScreen
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:7819]: nit: it's not clear what this example is describing

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 2]
```diff
@@ -2218,6 +2218,22 @@ public Color TransparencyKey
             }
         }
 
+        /// <summary>
+        /// Adjusts form location based on <see cref="FormStartPosition" >FormStartPosition</see>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs:2222]: `<see cref="FormStartPosition" >FormStartPosition</see>`
->
`<see cref="FormStartPosition"/>`

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.cs:2222]: Please add something about the DPI and Handle creation, for example copy/paste the PR description here.

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 3]
```diff
@@ -2218,6 +2218,22 @@ public Color TransparencyKey
...
+        {
+            FormStartPosition startPos = (FormStartPosition)formState[FormStateStartPos];
+            if (startPos == FormStartPosition.CenterParent)
+            {
+                CenterToParent();
+            }
+            else if (startPos == FormStartPosition.CenterScreen)
+            {
+                CenterToScreen();
+            }
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs:2234]: Start position could be defined as a `Location` property, do we assume that developer know what they are doing when setting location property?

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good, I added some nits

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs (hunk 3 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.cs:2234]: If user sets `StartPosition` and `Location `, they contradict each other. I assume developer knows how these properties to be used when changing from default values.

**Pull request merged**